core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/login/LoginViewModel.kt

```diff
@@ -156,10 +162,13 @@ class LoginViewModel(
         }
     }
 
-    fun onUsernameChanged(username: String) {
-        _uiState.update { prev ->
-            prev.copy(username = username)
-        }
+    fun onUsernameChanged(newValue: String) {
+        val filteredValue = filterUsernameUseCase(
```

Comment:
See above - we could avoid the extra .filter {} call by changing the invalidCharReplacement type to string, so then you can use ```invalidCharReplacement = ""```